src/passes/OptimizeInstructions.cpp

```diff
@@ -2543,23 +2543,52 @@ struct OptimizeInstructions
       }
     }
     {
-      // (i32(x) != 0) | (i32(y) != 0)   ==>   i32(x | y) != 0
-      // (i64(x) != 0) | (i64(y) != 0)   ==>   i64(x | y) != 0
+      Binary *bx, *by;
       Expression *x, *y;
+      Const *cx, *cy;
       if (matches(curr,
                   binary(OrInt32,
-                         binary(Ne, any(&x), ival(0)),
-                         binary(Ne, any(&y), ival(0)))) &&
-          x->type == y->type) {
-        auto* inner = curr->left->cast<Binary>();
-        inner->left = Builder(*getModule())
-                        .makeBinary(Abstract::getBinary(x->type, Or), x, y);
-        return inner;
+                         binary(&bx, any(&x), ival(&cx)),
+                         binary(&by, any(&y), ival(&cy)))) &&
+          bx->op == by->op && x->type == y->type && cx->value == cy->value &&
```

Comment:
I don't see a test for the corner cases of `cx->value != cy->value` or of `bx->op != by->op`.

Reply:
Sure! Done!